.github/workflows/reusable-workflow.yml

```diff
@@ -0,0 +1,46 @@
+name: Reusable Workflow
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        required: true
+        type: string
+      test-type:
+        required: true
+        type: string
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set Up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - name: Display Python Version
+        run: python -c "import sys; print(sys.version)"
+        shell: bash
+      - name: Install STUMPY And Other Dependencies
+        run: python -m pip install --editable .[ci]
+        shell: bash
+      - name: Run Black
+        run: black --check --diff ./
+        shell: bash
+      - name: Run Flake8
+        run: flake8 ./
+        shell: bash
+      - name: Link OpenMP
+        run: |
+             if [ "$RUNNER_OS" == "macOS" ]; then
+                  brew link --force libomp
+             fi
+        shell: bash
+      - name: Show Full Numba Environment
+        run: python -m numba -s
+        shell: bash
+      - name: Run Tests
+        run: ./test.sh ${{ inputs.test-type }}
+        shell: bash
```

Comment:
The two following steps are missing here:

```
- name: Check Coverage Report
        run: coverage report -m --fail-under=100 --skip-covered --omit=setup.py,docstring.py,min.py,stumpy/cache.py
        shell: bash
- name: Upload Coverage Tests Results
  uses: codecov/codecov-action@v4
``` 

So, we can first check If ${{ inputs.test-type }} is `coverage`, and, if yes, then run the steps above.